pkg/storage/image.go

```diff
@@ -99,18 +99,22 @@ func (svc *imageService) ListImages(systemContext *types.SystemContext, filter s
 			return nil, err
 		}
 		for _, image := range images {
-			ref, err := svc.getRef(image.Names[0])
-			if err != nil {
-				return nil, err
-			}
-			img, err := ref.NewImage(systemContext)
-			if err != nil {
-				return nil, err
+			var size *uint64
+			if len(image.Names) != 0 {
+				ref, err := svc.getRef(image.Names[0])
+				if err != nil {
```

Comment:
Should we log a warning if this happens? Is this a normal thing to happen?

Reply:
it's not I suppose, coming from bad state, maybe @nalind can help here?

Reply:
If the image never had a name assigned to it (e.g., it was built without being tagged), or the name that it had was subsequently moved to a different image, then it wouldn't have a name.

Reply:
but how do we get its size then if the only way to read the size is to create an image and call `imageSize` on it?

Reply:
Can we default the name to the imageid if there is no name specified, or would that break stuff?

Reply:
Looking at `storageTransport.ParseStoreReference`, it should already be possible to refer to unnamed images by ID.


> Can we default the name to the imageid if there is no name specified, or would that break stuff?

(Untargeted rant: it seems to me that the naming could benefit from someone _very precisely_ documenting what is the semantics of various strings used to refer to images, and making sure the code matches (i.e. generates only those formats, and accepts only those formats). E.g. this loop calls `svc.getRef`, which first tries `alltransports.ParseImageName`, i.e. this bit of code structure allows `imageService.ListImages` to return images from _completely different_ locations than `imageService` (including images on a remote registry) if the `imageService` contained an image with a specially formatted name, either by coincidence or intentionally.)

Reply:
So guys, the thing is, we need to find a way to get the image size w/o a name. We need this for kube and cadvisor. I can't think of a way of importing an image w/o a name other than some corruption, otherwise we pull images by name so I expect a name to be always available.
What we need to do though, is decoupling the image size from the image name.. This PR can be merged now, as prevents a panic but we still need to figure out image size w/o having a name. I'll create an issue later today.

Reply:
> I can't think of a way of importing an image w/o a name other than some corruption

`skopeo copy … containers-storage:@AA55AA…55`, or so. (Perhaps this is unsupported in context of cri-o, I don’t know.)(

ACK on preventing the panic for now, sure.

Reply:
> I can't think of a way of importing an image w/o a name other than some corruption

Actually, looking at `containers/storage/images.go`, probably just pull `foo:latest` once, and then pull `foo:latest` again after the `:latest` tag is moved to a different image: at least the `c/i/storage` transport will move the name but not delete the previous image.

Reply:
Well, I guess it's fine to have images w/o names. And as said, the real issue here is that we don't have a way to get image size w/o its name. We need that asap actually.